R/load-simulation.R

```diff
@@ -1,10 +1,52 @@
 #' Load a simulation from a pkml file and returns the simulation
 #'
 #' @param pkmlSimulationFile Full path of pkml simulation file to load.
+#' @param loadFromCache If TRUE, an already loaded pkml file will not be loaded
+#' again, but the simulation object will be retrieved from cache. This is the
+#' default behavior. If FALSE, new object will be created.
 #'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#'
+#' # Load sim1 for the first time
+#' sim1 <- loadSimulation(simPath)
+#'
+#' # sim2 will be loaded from cache and will represent the same object as sim1
+#' sim2 <- loadSimulation(simPath)
+#'
+#' parameter1 <- getParameter(toPathString(c("Organism", "Liver", "Volume")), sim1)
+#' parameter2 <- getParameter(toPathString(c("Organism", "Liver", "Volume")), sim2)
+#'
+#' # parameter1 and parameter2 belong to the same simulation object, so changing
+#' # one of the them will also change another
+#' setParametersValues(parameters = parameter2, values = 0)
+#' parameter1$value == parameter2$value # TRUE
+#'
+#' # sim3 will be loaded from not from cache
+#' sim3 <- loadSimulation(simPath, loadFromCache = FALSE)
+#' # parameter3 belong to different simulation object than parameter1 and parameter2
+#' parameter3 <- getParameter(toPathString(c("Organism", "Liver", "Volume")), sim3)
+#' setParametersValues(parameters = parameter3, values = 1)
+#' parameter2$value == parameter3$value # FALSE#'
 #' @export
-loadSimulation <- function(pkmlSimulationFile) {
+loadSimulation <- function(pkmlSimulationFile, loadFromCache = TRUE) {
+  validateIsOfType(loadFromCache, "logical")
+
+  if (loadFromCache) {
+    # If the file has already been loaded, return the last loaded object
+    if (exists(pkmlSimulationFile, where = ospsuiteEnv$loadedSimulations)) {
+      return(ospsuiteEnv$loadedSimulations[[pkmlSimulationFile]])
+    }
+  }
+
+  # If the simulation has not been loaded so far, or loadFromCache == FALSE,
+  # new simulation object will be created
   simulationLoader <- getNetTask("SimulationLoader")
-  netSim <- clrCall(simulationLoader, "LoadSimulation", pkmlSimulationFile)
-  Simulation$new(netSim)
+  netSim <- rClr::clrCall(simulationLoader, "LoadSimulation", pkmlSimulationFile)
+  simulation <- Simulation$new(netSim)
+
+  # Add the simulation to the cache of loaded simulations
+  ospsuiteEnv$loadedSimulations[[pkmlSimulationFile]] <- simulation

```

Comment:
It will always overwrite (if existing) the value assigned to a path when a simulation is loaded not from cache.